src/main/java/com/cocoapebbles/terraform/controllers/Controller.java

```diff
@@ -0,0 +1,8 @@
+package com.cocoapebbles.terraform.controllers;
+
+public interface Controller<SHAPE> {
```

Comment:
SHAPE is probably bad noun here.  This could technically be used for Entity/Schematics/etc